From 9746ade3c3e95e87301b5bed84abd829ead9aac7 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 19 Jan 2015 17:16:35 +0800
Subject: [PATCH] MLK-10116 tty: serial: imx: fix flush buffer issue when
 module clock is at 4Mhz

In general, uart module clock require it is great than 80Mhz to match 5Mbps
baud rate. When test below 14Mhz module clock, software reset cause state
machines off normal. And for i.MX6SL evk board low power test, it set uart
module clock to 4Mhz, which cause console port print out messy code.
The patch just is workaround to fix console issue.

Signed-off-by: Fugang Duan <B38611@freescale.com>
---
 drivers/tty/serial/imx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 313f0b3..3a760be 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1388,6 +1388,10 @@ static void imx_flush_buffer(struct uart_port *port)
 		dmaengine_terminate_all(sport->dma_chan_tx);
 	}
 
+	/* For console port, it is not necessary flush buffer and reset FIFO */
+	if (uart_console(port))
+		return;
+
 	/*
 	 * UCR2_SRST will reset the transmit and receive state machines,
 	 * all FIFOs and register UBIR, UBMR, UBRC,
-- 
1.8.0

